tsdb/store.go

```diff
@@ -155,6 +155,12 @@ func (s *Store) loadShards() error {
 				return err
 			}
 			for _, sh := range shards {
+				// Check to see if there are any `.tmp` directories that were left over from failed shard snapshots
+				if sh.IsDir() && strings.HasSuffix(sh.Name(), ".tmp") {
```

Comment:
No, there is already a `cleanup` func called when the engine is opened: 

See: https://github.com/influxdata/influxdb/blob/master/tsdb/engine/tsm1/engine.go#L150


and
https://github.com/influxdata/influxdb/blob/master/tsdb/engine/tsm1/engine.go#L789

Since the engine is creating the dir and defines the naming convention, seems like it should also be responsible for cleaning up.